cxx4opencl/diff2openclc.txt

```diff
@@ -120,10 +124,46 @@ label:  // invalid: jumping forward over declaration of n
    // ...
 ----------
 
-==== Use of Clang Blocks
+==== OpenCL specific difference
+
+This section described where {cpp} for OpenCL differs from OpenCL C
+in OpenCL specific behavior.
+
+===== Variadic macros
+
+{cpp} for OpenCL eliminates restriction on variadic macro from
+section 6.9.e of OpenCL C. Variadic macros can be used regularly
+as per {cpp}17 [cpp.replace].
+
+===== Predefined macros
+
+Predefined version macro `__OPENCL_C_VERSION__` and `NULL` described
+in OpenCL C 2.0 section 6.10 are not supported.
+
+There is a macro `__OPENCL_CPP_VERSION__` available that is set to
+value `100`. For convenience there is also `__CL_CPP_VERSION_1_0__`
+that is defined to the same value `100` and can be used instead of a
+literal.
+
+===== Atomic operations
+
+{cpp} for OpenCL relaxes restriction from OpenCL C 2.0 section 6.13.11 to
+atomic types by allowing to modify variables of such types using builtin
+operators, and not only using builtin function.
+
+The behavior of operators on atomic types follows the logic from {cpp}17
+sections [atomics.types.int][atomics.types.pointer][atomics.types.float].
+
+[source,c]
+----------
+atomic_int acnt;
+acnt++; // equivalent to atomic_fetch_add_explicit(&acnt, 1);
+----------
+
+===== Use of Clang Blocks
 
 Clang Blocks that are defined by the Objective-C language are not supported
 and their use can be replaced by lambdas ({cpp}17 `[expr.prim.lambda]`).
 
-This means that builtin functions using blocks, such as `enqueue_kernel`,
+This also implies that builtin functions using blocks, such as `enqueue_kernel`,
```

Comment:
Seems slightly unusual to start a paragraph with This also implies. Perhaps say something like "The above implies ... "